### PR TITLE
871811 - Added UG contents for MAUI TreeMap control

### PR DESCRIPTION
**Pull request by Kerubananthan3761**
### Description ###

Added UG content for the following topics for Maui TreeMap control:

### CODE REVIEW

File: MAUI/TreeMap/getting-started.md [hunk 1]
```diff
@@ -155,7 +155,11 @@ public class PopulationViewModel
 
 #### Bind data source for TreeMap
 
-To populate the treemap items, utilize the `DataSource` property of `SfTreeMap`. The `PrimaryValuePath` specifies the name of the property within the data object that provides the primary value used to determine the size of each item in the treemap. This primary value typically represents the main quantitative data associated with each item.
+To populate the treemap items, utilize the `DataSource` property of `SfTreeMap`. Along with `DataSourceThe`. Alongside, consider the follwing properties of `SfTreeMap` while initializing the tree map contol.
+* The `PrimaryValuePath` specifies the name of the property within the data object that provides the primary value used to determine the size of each item in the treemap. This primary value typically represents the main quantitative data associated with each item. 
+* The `RangeColorValuePath` specifies the name of the property within the data object that provides the range color value for each item in the tree map.This range color value typically represents a quantitative data range associated with each item, which is used to determine the color of the item based on a gradient or range mapping.
```
Comment (Karthickmani97) [inline MAUI/TreeMap/getting-started.md:160]: Mention legend and applicable for Treemap range.

File: MAUI/TreeMap/levels.md [hunk 2]
```diff
@@ -0,0 +1,681 @@
...
+layout: post
+title: Understanding Levels in .NET MAUI TreeMap | Syncfusion
+description: Learn about the levels within Syncfusion .NET MAUI TreeMap control, exploring its fundamental features and functionalities for effective treemap visualization.
+platform: maui
+control: TreeMap (SfTreeMap)
+documentation: ug
+---
+ 
+# Levels in .NET MAUI TreeMap (SfTreeMap)
+Levels in .NET MAUI TreeMap allow for hierarchical data visualization, enabling multi-level categorization and exploration of complex datasets. You can seamlessly navigate through different levels of data hierarchy, facilitating comprehensive understanding and analysis.
```
Comment (Karthickmani97) [inline MAUI/TreeMap/levels.md:11]: You can seamlessly navigate through different levels of data hierarchy, facilitating comprehensive understanding and analysis.

Seems like drill down support.

File: MAUI/TreeMap/levels.md [hunk 3]
```diff
@@ -0,0 +1,681 @@
...
+
+## TreeMap Level
```
Comment (Karthickmani97) [inline MAUI/TreeMap/levels.md:13]: rename to group path

File: MAUI/TreeMap/levels.md [hunk 4]
```diff
@@ -0,0 +1,681 @@
...
+Using `Levels` property of `SfTreeMap` you can integrate multiple hierarchical data levels to the TreeMap. The `GroupPath` property of `TreeMapLevel` specifies the name of the property in the data object that determines how items in the tree map are grouped. The grouping is based on the values of this property in the data source, which splits the level into distinct groups.
+
```
Comment (Karthickmani97) [inline MAUI/TreeMap/levels.md:15]: In the following example, three levels are added and each level is configured using the GroupPath.

File: MAUI/TreeMap/levels.md [hunk 5]
```diff
@@ -0,0 +1,681 @@
...
+    {
+        get;
+        set;
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+## Level customization
```
Comment (Karthickmani97) [inline MAUI/TreeMap/levels.md:106]: Customize level appearance

File: MAUI/TreeMap/levels.md [hunk 6]
```diff
@@ -0,0 +1,681 @@
...
+The .NET MAUI TreeMap control allows you to customize the background, text style, spacing, and more.
```
Comment (Karthickmani97) [inline MAUI/TreeMap/levels.md:107]: The level appearance customization can be achieved by using the `TextStyle`, `Background`, `Spacing`, `Stroke,` and `StrokeWidth` properties of `TreeMapLevel` in the `SfTreeMap.`

File: MAUI/TreeMap/levels.md [hunk 7]
```diff
@@ -0,0 +1,681 @@
...
+
+### Customize the levels spacing
```
Comment (Karthickmani97) [inline MAUI/TreeMap/levels.md:109]: Customize level appearance using spacing

File: MAUI/TreeMap/levels.md [hunk 8]
```diff
@@ -0,0 +1,681 @@
...
+Customize the spacing between the tree map header items using the `Spacing` property of `TreeMapLevel`. The default value of `Spacing` property is `1`.
```
Comment (Karthickmani97) [inline MAUI/TreeMap/levels.md:110]: The `Spacing` property can be customized by using the `TreeMapLevel` of `SfTreeMap.` It is used to customize the the spacing between the tree map header items and its default value of `Spacing` property is `1.`

File: MAUI/TreeMap/selection-and-highlights.md [hunk 9]
```diff
@@ -0,0 +1,548 @@
...
+title: Selection and highlights in .NET MAUI TreeMap | Syncfusion
+description: Learn about the functionality of selection and highlights within the Syncfusion .NET MAUI TreeMap(STreeMap) control.
+platform: maui
+control: TreeMap (SfTreeMap)
+documentation: ug
+---
+ 
+# Selection in TreeMap control
+
+The Treemap control allows interactive selection of items by clicking or tapping. `Tooltip`, if enabled, displays additional information upon clicking or tapping, enhancing user understanding of data points. When `SelectionMode` is `None`, interaction is disabled for treemap items.
```
Comment (Karthickmani97) [inline MAUI/TreeMap/selection-and-highlights.md:12]: The Treemap control allows to select a specific group or item to distinguish it from others. When enabled, the Tooltip feature displays additional information during interaction. However, when SelectionMode is set to None, interaction is disabled for treemap items.

File: MAUI/TreeMap/selection-and-highlights.md [hunk 10]
```diff
@@ -0,0 +1,548 @@
...
+
+## Selection mode
+
+Selection mode in the Treemap control determines the interactivity level, enabling users to choose specific items by clicking or tapping. The default value of `SelectionMode` is `SelectionMode.Single`.
+
+### Single selection
+With `Single` selection mode, you can choose one item at a time within the Treemap control, ideal for isolating and examining individual data points thoroughly.
```
Comment (Karthickmani97) [inline MAUI/TreeMap/selection-and-highlights.md:19]: In the Single selection mode, users can choose one item at a time within the Treemap control.

File: MAUI/TreeMap/selection-and-highlights.md [hunk 11]
```diff
@@ -0,0 +1,548 @@
...
+        get;
+        set;
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+### Multiple selection
+With `Multiple` selection mode you can select multiple items simultaneously, facilitating comparative analysis and exploration of related data within the Treemap.
```
Comment (Karthickmani97) [inline MAUI/TreeMap/selection-and-highlights.md:108]: In the Multiple selection mode, users can select multiple items simultaneously within the Treemap control.

File: MAUI/TreeMap/selection-and-highlights.md [hunk 12]
```diff
@@ -0,0 +1,548 @@
...
+        get;
+        set;
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+### None
+When set to none selection mode, interactivity within the Treemap control is disabled.
```
Comment (Karthickmani97) [inline MAUI/TreeMap/selection-and-highlights.md:197]: In the None selection mode, interaction with treemap items is disabled.

File: MAUI/TreeMap/levels.md [hunk 13]
```diff
@@ -0,0 +1,689 @@
...
+        set;
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+### Customize level appearance using background
+
+The `Background` property of `TreeMapLevel` can be customized by using the `Levels` in the `SfTreeMap.` It is used to customize the background brush for the tree map header item. The default value of `Background` is `Null.`
```
Comment (M-A32) [inline MAUI/TreeMap/levels.md]: The `Background` property of `TreeMapLevel` can be customized by using the `Levels` in the `SfTreeMap.` It is used to customize the background brush for the treemap header item. The default value of `Background` is `Null.`

File: MAUI/TreeMap/levels.md [hunk 14]
```diff
@@ -0,0 +1,689 @@
...
+    {
+        get;
+        set;
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+N> Setting a background value for the property applies the same background color to tree map header items within the same level. To customize the background color for both header and leaf items at each level, utilize the `GroupItemBrushSettings` property.
```
Comment (M-A32) [inline MAUI/TreeMap/levels.md]: N> Setting a background value for the property applies the same background color to treemap header items within the same level. To customize the background color for both header and leaf items at each level, utilize the `GroupItemBrushSettings` property.

File: MAUI/TreeMap/levels.md [hunk 15]
```diff
@@ -0,0 +1,689 @@
...
+
+### Customize level appearance using stroke
+
+The `Stroke` property of `TreeMapLevel` can be customized by using the `Levels` in the `SfTreeMap.` It is used to customize the stroke color for the tree map header item. The default value of `Stroke` is `new SolidColorBrush(Color.FromArgb("#CAC4D0")).`
```
Comment (M-A32) [inline MAUI/TreeMap/levels.md]: The `Stroke` property of `TreeMapLevel` can be customized using the `Levels` in the `SfTreeMap.` It is used to customize the stroke color for the treemap header item. The default value of `Stroke` is `new SolidColorBrush(Color.FromArgb("#CAC4D0")).`

File: MAUI/TreeMap/levels.md [hunk 16]
```diff
@@ -0,0 +1,689 @@
...
+        set;
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+###  Customize level appearance using stroke width
+
+The `StrokeWidth` property of `TreeMapLevel` can be customized by using the `Levels` in the `SfTreeMap.` It is used to customize the width of the stroke for the tree map header item. The default value of `StrokeWidth` is `1.`
```
Comment (M-A32) [inline MAUI/TreeMap/levels.md]: The `StrokeWidth` property of `TreeMapLevel` can be customized by using the `Levels` in the `SfTreeMap.` It is used to customize the width of the stroke for the treemap header item. The default value of `StrokeWidth` is `1.`

File: MAUI/TreeMap/levels.md [hunk 17]
```diff
@@ -0,0 +1,689 @@
...
+        set;
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+### Customize level appearance using style
+
+The `TextStyle` property of `TreeMapLevel` can be customized by using the `Levels` in the `SfTreeMap.` It is used to customize the `TextColor,` `FontSize,` `FontFamily` and `FontAttributes.` of tree map header item text.
```
Comment (M-A32) [inline MAUI/TreeMap/levels.md:591]: The `TextStyle` property of `TreeMapLevel` can be customized using the `Levels` in the `SfTreeMap.` It is used to customize the `TextColor,` `FontSize,` `FontFamily` and `FontAttributes.` of treemap header item text.

File: MAUI/TreeMap/selection-and-highlights.md [hunk 18]
```diff
@@ -0,0 +1,548 @@
+---
+layout: post
+title: Selection and highlights in .NET MAUI TreeMap | Syncfusion
```
Comment (M-A32) [inline MAUI/TreeMap/selection-and-highlights.md:3]: title: Selection and Highlights in .NET MAUI TreeMap | Syncfusion

File: MAUI/TreeMap/selection-and-highlights.md [hunk 19]
```diff
@@ -0,0 +1,548 @@
...
+documentation: ug
+---
+ 
+# Selection in TreeMap control
+
+The Treemap control allows interactive selection of items by clicking or tapping. `Tooltip`, if enabled, displays additional information upon clicking or tapping, enhancing user understanding of data points. When `SelectionMode` is `None`, interaction is disabled for treemap items.
+
+## Selection mode
+
+Selection mode in the Treemap control determines the interactivity level, enabling users to choose specific items by clicking or tapping. The default value of `SelectionMode` is `SelectionMode.Single`.
```
Comment (M-A32) [inline MAUI/TreeMap/selection-and-highlights.md:16]: The selection mode in the Treemap control determines the interactivity level, enabling users to choose specific items by clicking or tapping. The default value of `SelectionMode` is `SelectionMode.Single`.

File: MAUI/TreeMap/selection-and-highlights.md [hunk 20]
```diff
@@ -0,0 +1,548 @@
...
+        get;
+        set;
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+## Select items programatically
+You can programmatically add default selected items by utilizing the `SelectedItems` property of `SfTreeMap`. This property holds a collection of tree map items representing the currently selected ones. In `Single` selection mode, only the first item in the SelectedItems collection will be selected. In `Multiple` selection mode, all items in the SelectedItems collection will be selected. When `SelectionMode` is set to `None`, no items will be selected automatically.
```
Comment (M-A32) [inline MAUI/TreeMap/selection-and-highlights.md:286]: You can programmatically add default selected items by utilizing the `SelectedItems` property of `SfTreeMap`. This property holds a collection of tree map items representing the currently selected ones. In the `Single` selection mode, only the first item in the SelectedItems collection will be selected. In the `Multiple` selection mode, all items in the SelectedItems collection will be selected. When `SelectionMode` is set to `None`, no items will be selected automatically.

File: MAUI/TreeMap/selection-and-highlights.md [hunk 21]
```diff
@@ -0,0 +1,548 @@
...
+    {
+        get;
+        set;
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+## Customize selected item border color
```
Comment (Karthickmani97) [inline MAUI/TreeMap/selection-and-highlights.md:372]: Customize selected item appearence

The selected items appearance customization can be achieved by using the SelectedItemStroke and SelectedItemStrokeWidth properties in the SfTreeMap.

File: MAUI/TreeMap/selection-and-highlights.md [hunk 22]
```diff
@@ -0,0 +1,548 @@
...
+Customize the selected item border color using `SelectedItemStroke` property of `SfTreeMap`. The default value of `SelectedItemStroke` is `Color.FromArgb("#1C1B1F")`.
```
Comment (M-A32) [inline MAUI/TreeMap/selection-and-highlights.md:373]: Customize the selected item border color using the `SelectedItemStroke` property of `SfTreeMap`. The default value of `SelectedItemStroke` is `Color.FromArgb("#1C1B1F")`.

File: MAUI/TreeMap/selection-and-highlights.md [hunk 23]
```diff
@@ -0,0 +1,548 @@
...
+        get;
+        set;
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+## Customize selected item border thickness
+Customize selected item border thickness using `SelectedItemStrokeWidth` property of `SfTreeMap`. The default value of `SelectedItemStrokeWidth` is `2`.
```
Comment (M-A32) [inline MAUI/TreeMap/selection-and-highlights.md:462]: Customize selected item border thickness using the `SelectedItemStrokeWidth` property of `SfTreeMap`. The default value of `SelectedItemStrokeWidth` is `2`.

**Review by M-A32 — CHANGES_REQUESTED** [review]
I have completed the review.

File: MAUI/TreeMap/leaf-item-settings.md [hunk 24]
```diff
@@ -0,0 +1,551 @@
...
+layout: post
+title: Leaf item settings in .NET MAUI TreeMap | Syncfusion
+description: Learn how to harness Syncfusion .NET MAUI TreeMap(STreeMap) control leaf item settings for precise fill color customization, enhancing visual appeal.
+platform: maui
+control: TreeMap (SfTreeMap)
+documentation: ug
+---
+ 
+# TreeMap Leaf item settings
+The `LabelPath` property within the `TreeMapLeafItemSettings` is used to display text for the leaf items. This property allows you to specify the path to the data property containing the text you want to display. The `TreeMapLeafItemSettings` also allows you to customize the spacing between items, stroke color, stroke width, text style, and text format option in the TreeMap control.
```
Comment (Karthickmani97) [inline MAUI/TreeMap/leaf-item-settings.md:11]: A leaf item represents a visualized data element in the TreeMap. It does not contain child items but may have a parent item if levels are specified in the TreeMap control.

**Review by karthikraja-arumugam — APPROVED** [review]

**Pull request merged**